.github/workflows/png_file_checks.yml

```diff
@@ -0,0 +1,27 @@
+name: png_file_checks
+
+# Check whether all png files are in a valid format
+on:
+  push:
+    paths:
+      - '**.png'
+      - '.github/workflows/**.yml'
+  pull_request:
+    paths:
+      - '**.png'
+      - '.github/workflows/**.yml'
+
+jobs:
+  png_optimized:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install deps
+        run: |
+          sudo apt-get update
+          sudo apt install -y optipng
+
+      # Does only run a fuzzy check without -o7 -zm1-9 since it would be too slow otherwise
+      - name: Check whether all png files are optimized
+        run: |
+          git ls-files *.png | xargs -I % bash -c "optipng -nc -strip all -simulate % 2>&1 >/dev/null | (if ! grep -q 'is already optimized' $1 ; then echo -e '\033[0;31m%\033[0m'; exit 1; fi)" || (echo "Found non-optimized png file, apply 'optipng -o7 -zm1-9 -nc -strip all -clobber <filename>'"; exit 1;)
```

Comment:
please move this to a script file. there is no need to code-golf this into a single line

Reply:
Done